sw/device/lib/testing/test_framework/ottf_isrs.c

```diff
@@ -0,0 +1,119 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/test_framework/ottf_isrs.h"
+
+#include "sw/device/lib/base/csr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+
+static uint32_t get_mtval(void) {
+  uint32_t mtval;
+  CSR_READ(CSR_REG_MTVAL, &mtval);
+  return mtval;
+}
+
+__attribute__((weak)) void ottf_exception_handler(void) {
+  uint32_t mcause;
+  ottf_exc_id_t exc_cause;
+
+  CSR_READ(CSR_REG_MCAUSE, &mcause);
+  exc_cause = (ottf_exc_id_t)(mcause & kIdMax);
+
+  switch (exc_cause) {
+    case kInstrMisaligned:
+      ottf_instr_misaligned_fault();
+      break;
+    case kInstrAccessFault:
+      ottf_instr_access_fault();
+      break;
+    case kIllegalInstrFault:
+      ottf_illegal_instr_fault();
+      break;
+    case kBreakpoint:
+      ottf_breakpoint();
+      break;
+    case kLoadAccessFault:
+      ottf_load_store_fault();
+      break;
+    case kStoreAccessFault:
+      ottf_load_store_fault();
+      break;
+    case kMachineECall:
+      ottf_machine_ecall();
+      break;
+    case kUserECall:
+      ottf_user_ecall();
+      break;
+    default:
+      LOG_INFO("Unknown exception triggered!");
+      while (1) {
```

Comment:
`s/1/true`. The trailing semi is superfluous.

Also, make sure to include a `wait_for_interrupt()` in the loop body; infinite empty loops are UB and Clang will delete it.

Applies throughout. (Maybe worth defining a `static noreturn void loop_forever();`...)

Reply:
Fixed.